lib/WeBWorK/PG/Environment.pm

```diff
@@ -58,16 +58,14 @@ use YAML::XS qw/LoadFile/;
 sub new ($invocant, $courseName = '___') {
 	my $class = ref $invocant || $invocant;
 
-	my $pg_root;
+	my $pg_root = $ENV{PG_ROOT};
+	die 'The pg directory must be defined in PG_ROOT.' unless -d $pg_root;
+
 	my $ce;
 
 	eval {
 		require WeBWorK::CourseEnvironment;
-		$ce      = WeBWorK::CourseEnvironment->new({ webwork_dir => $ENV{WEBWORK_ROOT}, courseName => $courseName });
-		$pg_root = $ce->{pg_root};
-	} or do {
-		$pg_root = $ENV{PG_ROOT};
-		die 'The pg directory must be defined in PG_ROOT.' unless -d $pg_root;
+		$ce = WeBWorK::CourseEnvironment->new({ webwork_dir => $ENV{WEBWORK_ROOT}, courseName => $courseName });
 	};
```

Comment:
Does this block still need to be in an `eval`?  Is there anyway for this block to fail.  I thought that was generally what `eval` was for?

Reply:
It certainly needs to be an `eval` still.  The standalone renderer will not have the `WeBWorK::CourseEnvironment` module, and so the code in the eval block will fail for the standalone renderer.  That failure will quietly be ignored, because that is what is supposed to happen.  Without the `eval` block there would be an error that would make the code cease to function.